src/style.rs

```diff
@@ -674,12 +678,98 @@ enum Alignment {
     Right,
 }
 
+/// Trait for defining stateful or stateless formatters
+pub trait ProgressTracker: Send {
+    fn cloned(&self) -> Box<dyn ProgressTracker>;
```

Comment:
On second thought, what if we called it derive() or derived()? I think that would pretty succinctly describe what it does.

Reply:
It's not an obvious improvement to me... @chris-laplante what do you think?

Also would be good to add some documentation to these trait method definitions.

Reply:
Yup, let's do clone_box!

Reply:
All set